orchestrator/domain/base.py

```diff
@@ -387,13 +387,15 @@ def _data_from_lifecycle(cls, other: "DomainModel", status: SubscriptionLifecycl
                         possible_product_block_types[item.name]._from_other_lifecycle(item, status, subscription_id)
                     )
             else:
-                if is_optional_type(field_type):
+                if is_optional_type(field_type) and not is_union_type(field_type):
                     data[field_name] = None
                     field_type = first(get_args(field_type))
                 elif is_union_type(field_type):
+                    if is_optional_type(field_type):
+                        data[field_name] = None
                     field_types = get_args(field_type)
                     for f_type in field_types:
-                        if f_type.name == value.name:
+                        if value and not isinstance(None, f_type) and f_type.name == value.name:
```

Comment:
`not isinstance(None, f_type)` can None be subclassed?

Reply:
`f_type` can be a `NoneType`, don't know a better way to validate that

Reply:
And `if value` is not ideal for falsy values

Reply:
> `f_type` can be a `NoneType` class, don't know a better way to validate that

Ah, I see that `types.NoneType` was added in 3.10. Yeah, makes sense then :)